src/radical/entk/appman/appmanager.py

```diff
@@ -298,13 +316,36 @@ def resource_desc(self, value):
             self._logger.error('Could not validate resource description')
             raise EnTKError('Could not validate resource description')
 
+        self._resource_desc = value
+
         self._rmgr._populate()
         self._rmgr.shared_data = self._shared_data
         self._rmgr.outputs     = self._outputs
 
+        if self._services:
+            self._rmgr.services = self._services
+
         self._report.ok('>>ok\n')
 
 
+    # --------------------------------------------------------------------------
+    #
+    @services.setter
+    def services(self, tasks):
+
+        if not isinstance(tasks, list):
+            tasks = [tasks]
+
+        for task in tasks:
+            if not isinstance(task, Task):
+                raise EnTKTypeError(expected_type=Task,
+                                    actual_type=type(task))
+
+        self._services = tasks
+        if self._rmgr:
+            self._rmgr.services = tasks
```

Comment:
It's probably outside of the scope of this PR, but I want to mention it.  That pattern to keep the data as member attribute or as rmgr attribute, depending on the call order, is really confusing.  I think data should be kept as member attributes in general, and the rmgr should only be created on `self.run()`, and only then get assigned the attributes all at once.  That would simplify several attribute setters and also make error modes more predictable.

But, as said, out of scope for here IMHO...

Reply:
agree, and I didn't want to touch that part at this PR